server/middleware/tchannel-client/helpers/with-next-page-token-request-transform.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021 Uber Technologies Inc.
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+const withNextPageTokenRequestTransform = body => {
+  if (!body || typeof body !== 'object') {
+    return body;
+  }
+
+  const { nextPageToken } = body;
+
+  body.nextPageToken = nextPageToken
+    ? Buffer.from(nextPageToken, 'base64')
```

Comment:
tbh I'm surprised this is necessary at all... normally I'd think we leave it alone and things work, since the token isn't actually base64, though it contains mostly base64-encoded data.

does the javascript tchannel client convert binary fields to base64 strings?  that'd seem weird to me, but I guess it's possible

Reply:
This field comes through as a binary stream. It may be possible to not convert at all (will need to double check), but if we were to send this npt to the UI it would need to be converted to string as json (via REST) won't allow binary streams AFAIK. Let me do some testing and will evaluate whether we need this change or not. Thanks!

Reply:
If I disable both

Reply:
It looks like if I only enable bodyTransform (and have responseTranform disabled) it will work. I will see what other effects this has on tests etc.

Reply:
So my guess here is if the binary stream isn't converted, it will automatically convert to a base64 string when reading from the API, however if we don't have a body transform, the request will not be able to handle the base64 string and fail. I have updated the code and reverted some of the test changes as they are no longer necessary.